prometheus/exporter/intervalpush/exporter.go

```diff
@@ -23,20 +23,56 @@ type exporter struct {
 }
 
 func NewExporter(url string, interval time.Duration) exp.Exporter {
-	if interval == 0 {
-		interval = defaultInterval
+	return NewExporterWithOptions(
+		url,
+		WithInterval(interval),
```

Comment:
should we keep the previous behavior to ensure that the interval is not `0`  ?

Reply:
It creates the ticker with the default value. https://github.com/upfluence/pkg/blob/155324188ef8c92226a27423a576b8567e1aa4b0/prometheus/exporter/intervalpush/exporter.go#L62

Reply:
Nay, we could still pass an interval [here](https://github.com/upfluence/pkg/blob/155324188ef8c92226a27423a576b8567e1aa4b0/prometheus/exporter/intervalpush/exporter.go#L25).

we should keep this guard, otherwise it's a breaking change :)